src/device/socket/vsock.rs

```diff
@@ -108,7 +109,7 @@ pub struct VirtIOSocket<H: Hal, T: Transport> {
     /// The guest_cid field contains the guest’s context ID, which uniquely identifies
     /// the device for its lifetime. The upper 32 bits of the CID are reserved and zeroed.
     guest_cid: u64,
-    rx_buf_dma: Dma<H>,
+    rx_buf: NonNull<[u8; PAGE_SIZE]>,
```

Comment:
Why use `NonNull` instead of `Box` here?

Reply:
`new` and `pop_packet_from_rx_queue` add subranges of it to the RX queue, so it is not safe to access it at the same time. Using a `Box` would allow safe code to access it, which would be unsound. Using a `NonNull` instead requires an `unsafe` block for any code which accesses it.